superset/views/database/validators.py

```diff
@@ -53,3 +55,21 @@ def schema_allows_csv_upload(database, schema):
         security_manager.database_access(database)
         or security_manager.all_datasource_access()
     )
+
+
+def certificate_validator(
+    certificate: Optional[str], exception: Type[ValidationError] = ValidationError
+) -> None:
+    if certificate:
+        try:
+            parse_ssl_cert(certificate)
+        except CertificateException:
```

Comment:
I think we could change the `CertificateException` class and add the message property with this, then here just let `parse_ssl_cert` fail

Reply:
forget it just checked and the Exception must be a child of `ValidationError`.  We could make it a child of `ValidationError` and set the message on a constructor override